airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -120,3 +130,40 @@ def get_dataset_events(
     return dataset_event_collection_schema.dump(
         DatasetEventCollection(dataset_events=events, total_entries=total_entries)
     )
+
+
+@security.requires_access([(permissions.ACTION_CAN_CREATE, permissions.RESOURCE_DATASET)])
+@provide_session
+def post_dataset_event(session: Session = NEW_SESSION) -> APIResponse:
+    """Create an external dataset event. This endpoint is useful if you want to update a dataset and
+    trigger downstream DAG runs from external services.
+    """
+    try:
+        json_body = dataset_change_schema.load(get_json_request_dict())
+    except ValidationError as err:
+        raise BadRequest(detail=str(err))
+    uri = json_body["dataset_uri"]
+    external_source = request.remote_addr
```

Comment:
i am not sure that remote_addr is the right choice here.  maybe we could record such information in the log table?  but to me it would seem it might be more useful to let this be an arbitrary text field?  although i suppose user can always supply information in the `extra` dict.... wdyt?